stellar/contracts/wormhole-contract/src/utils/bytes_reader.rs

```diff
@@ -0,0 +1,181 @@
+use soroban_sdk::{Bytes, BytesN, Env};
+use wormhole_interface::Error;
+
+/// A cursor-based reader for parsing bytes in big-endian format.
+/// Tracks position and provides clean error handling for out-of-bounds reads.
+pub(crate) struct BytesReader<'a> {
```

Comment:
You can remove `(crate)` and mark the functions as`pub` only. Changing the visibility of the `utils` from `pub mod` to `mod` achieves the same result and hides `utility`. Currently, the package is empty for external users.